js/index_controls.js

```diff
@@ -79,8 +79,28 @@ function calculate() {
 				: Math.floor(p2.maxHP * p1.moves[i].hasRecoil * 0.48 / p1.maxHP);
 			}
 			recoilText = ' (' + minRecoilDamage + ' - ' + maxRecoilDamage + notation + ' recoil damage)';
+		} else if (p1.moves[i].hasRecoil === 'crash') {
+			var minRecoilDamage = notation === '%' ? Math.floor(Math.min(minDamage, p2.maxHP) * 10 / p1.maxHP) :
+				Math.floor(Math.min(minDamage, p2.maxHP) * 0.48 / p1.maxHP);
+			var maxRecoilDamage = notation === '%' ? Math.floor(Math.min(maxDamage, p2.maxHP) * 10 / p1.maxHP) :
+				Math.floor(Math.min(maxDamage, p2.maxHP) * 0.48 / p1.maxHP);
+			if (minDamage > p2.curHP) {
```

Comment:
Not really sure what this part is about, can you explain?

Reply:
Shouldn't all the checks for the opponent's maxHP be for curHP then?

Reply:
The checks for normal recoil moves, I mean.

Reply:
Yeah I thought the same thing I just didn’t get a chance to test that and make sure it works. Should be done tomorrow

Reply:
Sounds good. Anyway, I think the way to calculate crash damage would be just to take the full min/max damage and cap it depending on the gen and/or the user's actual HP, then deal with the fancy notation bits after. Am I missing something?